packages/esm-appointments-app/src/hooks/useClinicalMetrics.ts

```diff
@@ -37,45 +36,59 @@ export const useClinicalMetrics = () => {
   };
 };
 
-export function useAllAppointmentsByDate() {
+export const useAppointmentsForDate = () => {
```

Comment:
This shared hook abstracts away the API call to `/ws/rest/v1/appointment/all?forDate=${selectedDate}` and gets reused in `useAllAppointmentsByDate` and `useScheduledAppointments`.